en/ros/mavros_custom_messages.md

```diff
@@ -0,0 +1,348 @@
+   ...
+   void
+   MavlinkReceiver::handle_message_key_command(mavlink_message_t *msg)
+   {
+       mavlink_key_command_t man;
+       mavlink_msg_key_command_decode(msg, &man);
+
+   struct key_command_s key = {};
+
+       key.timestamp = hrt_absolute_time();
+       key.cmd = man.command;
+
+       if (_key_command_pub == nullptr) {
+           _key_command_pub = orb_advertise(ORB_ID(key_command), &key);
+
+       } else {
+           orb_publish(ORB_ID(key_command), _key_command_pub, &key);
+       }
+   }
+   ```
+
+1. Make your own uORB topic subscriber just like any example subscriber module. In my case, I made my own module in (/Firmware/src/inrol/key_receiver). 
+   In this directory, I have two files **CMakeLists.txt**, **key_receiver.cpp**.
+   Each one looks like following.
+
+   -CMakeLists.txt
+
+   ```cmake
+   px4_add_module(
+       MODULE inrol__key_receiver
+       MAIN key_receiver
+       STACK_MAIN 2500
+       STACK_MAX 4000
+       SRCS
+           key_receiver.cpp
+       DEPENDS
+           platforms__common
+
+       )
+   ```
+
+   -key_receiver.cpp
+
+   ```
+   #include <px4_config.h>
+   #include <px4_tasks.h>
+   #include <px4_posix.h>
+   #include <unistd.h>
+   #include <stdio.h>
+   #include <poll.h>
+   #include <string.h>
+   #include <math.h>
+
+   #include <uORB/uORB.h>
+   #include <uORB/topics/key_command.h>
+
+   extern "C" {__EXPORT int key_receiver_main(int argc, char **argv);}
+
+   int key_receiver_main(int argc, char **argv)
+   {
+       int key_sub_fd = orb_subscribe(ORB_ID(key_command));
+       orb_set_interval(key_sub_fd, 200); // limit the update rate to 200ms
+
+       px4_pollfd_struct_t fds[1];
+       fds[0].fd = key_sub_fd, fds[0].events = POLLIN;
+
+       int error_counter = 0;
+
+       while(1)
+       {
+           int poll_ret = px4_poll(fds, 1, 1000);
+
+           if (poll_ret == 0)
+           {
+               PX4_ERR("Got no data within a second");
+           }
+
+           else if (poll_ret < 0)
+           {
+               if (error_counter < 10 || error_counter % 50 == 0)
+               {
+                   PX4_ERR("ERROR return value from poll(): %d", poll_ret);
+               }
+
+               error_counter++;
+           }
+
+           else
+           {
+               if (fds[0].revents & POLLIN)
+               {
+                   struct key_command_s input;
+                   orb_copy(ORB_ID(key_command), key_sub_fd, &input);
+                   PX4_INFO("Recieved Char : %c", input.cmd);
+                }
+           }
+       }
+       return 0;
+   }
+   ```
+
+1. Add your module in **nuttx_px4fmu-v2_default.cmake** (in **Firmware/cmake/configs**).
+   ```cmake
+   set(config_module_list
+   ...
+   inrol/key_receiver
+   )
+   ```
+
+Now you are ready to build all your work!
+
+## Building
+
+### Build for ROS
+
+1. In your workspace enter: `catkin build`.
+1. Beforehand, you have to set your "px4.launch" in (/workspace/src/mavros/mavros/launch). 
```

Comment:
This part doesn't have anything to do with creating a new mavros plugin.. It's how to connect mavros and the flight controller

Reply:
@RicardoM17 Note these suggestions. If you have questions, I can help address.